src/pygeoprocessing/geoprocessing.py

```diff
@@ -1798,19 +1798,24 @@ def reclassify_raster(
     """Reclassify pixel values in a raster.
 
     A function to reclassify values in raster to any output type. By default
-    the values except for nodata must be in ``value_map``.
+    the values except for nodata must be in ``value_map``. If the base
+    raster nodata value is in ``value_map``, then nodata values are
+    reclassified to the mapped value. This would essentially leave no nodata
+    holes in the target raster.
```

Comment:
How about adding something like, "Include the base raster nodata value in ``value_map`` to reclassify nodata pixels to a valid value (not setting this as the target raster's nodata value). Set ``target_nodata`` to reclassify nodata pixels to a different nodata value (setting this as the target raster's nodata value)."

Reply:
I gave it a go here: https://github.com/natcap/pygeoprocessing/pull/221/commits/31b9c7164353426e15d7c43d801d3e5823038e57

>Set target_nodata to reclassify nodata pixels to a different nodata value (setting this as the target raster's nodata value)

I left off this last bit because `target_nodata` is a unique argument and `setting` it didn't quite feel like the right phrasing to me. I also think this part should be plenty explained in the Args section. Let me know what you think.

Reply:
That looks good to me!